### PR TITLE
Bid tracking alerts

### PR DESCRIPTION
**Pull request by mjoyce91**
Adds overlay alerts to the bid tracker steps. These are used for certain bid statuses where instead of displaying the tracker, a message or prompt is displayed to the user.

Covers #1115, #1117, #1118

Relies on #1132

### CODE REVIEW

File: src/Components/BidListResultsCard/BidStatus/BidStatus.jsx [hunk 1]
```diff
@@ -28,6 +28,10 @@ const BidStatus = ({ status }) => {
       icon = 'handshake-o';
       text = HAND_SHAKE_ACCEPTED.text;
       break;
+    case HAND_SHAKE_DECLINED.property:
```
Comment (burgwyn) [inline src/Components/BidListResultsCard/BidStatus/BidStatus.jsx]: Missing a test for this case

File: src/Components/BidTracker/BidHelpers.js [hunk 2]
```diff
@@ -162,6 +174,7 @@ function bidClassesFromCurrentStatus(bid = { status: 'draft' }) {
     // the tracker the same as the in-panel phase.
     case IN_PANEL_PROP:
     case APPROVED_PROP:
+    case DECLINED_PROP:
     case 'Cherries':
```
Comment (burgwyn) [inline src/Components/BidTracker/BidHelpers.js]: 🍒

File: src/Components/BidTracker/BidTrackerCard/BidTrackerCard.jsx [hunk 3]
```diff
@@ -1,42 +1,58 @@
...
           </div>
+          <div className="usa-grid-full padded-container-inner bid-tracker-bid-steps-container">
+            <BidSteps bid={bid} />
+            {
+              showAlert &&
+                <OverlayAlert
+                  id={bid.id}
+                  type={bid.status}
+                  userName={bid.user}
+                  bureau="(AF) Bureau of African Affairs"
```
Comment (burgwyn) [inline src/Components/BidTracker/BidTrackerCard/BidTrackerCard.jsx]: Will we get this from the API soon?

File: src/Components/BidTracker/OverlayAlert/ApprovedAlert/ApprovedAlert.jsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+const ApprovedAlert = ({ userName }) => (
+  <div className="bid-tracker-alert-container bid-tracker-alert-container--approved">
+    <div className="top-text">
+      {`${userName}, you've been `}<strong>approved</strong> for this position!
+      </div>
+    <div>
+      <button className="tm-button-transparent">Prepare for Next Steps</button>
+    </div>
+    <div className="sub-text">
```
Comment (burgwyn) [inline src/Components/BidTracker/OverlayAlert/ApprovedAlert/ApprovedAlert.jsx]: Let's add the StaticWrapper here and add a TODO for the travel link

File: src/Components/BidTracker/OverlayAlert/HandshakeOfferedAlert/HandshakeOfferedAlert.test.jsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    );
+    expect(wrapper).toBeDefined();
+  });
+
+  it('can accept a bid', () => {
+    const spy = sinon.spy();
+    const wrapper = shallow(
+      <HandshakeOfferedAlert id={1} userName="test" acceptBid={spy} declineBid={() => {}} />,
+    );
+    const button = wrapper.find('button').at(0);
```
Comment (burgwyn) [inline src/Components/BidTracker/OverlayAlert/HandshakeOfferedAlert/HandshakeOfferedAlert.test.jsx]: 👍

File: src/Components/BidTracker/OverlayAlert/InPanelAlert/InPanelAlert.jsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import FontAwesome from 'react-fontawesome';
+
+const InPanelAlert = ({ userName, bureau }) => (
+  <div className="bid-tracker-alert-container bid-tracker-alert-container--in-panel">
+    <div className="top-text">
+      <div>
+        <FontAwesome name="users" />
+      </div>
+      <div>
+        {`${userName}, you are in panel for your bid for ${bureau}`}
```
Comment (burgwyn) [inline src/Components/BidTracker/OverlayAlert/InPanelAlert/InPanelAlert.jsx]: I don't love the language here.  Can you confirm with @krisdelacruz?

File: src/Components/BidTracker/OverlayAlert/InPanelAlert/InPanelAlert.test.jsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  it('is defined', () => {
+    const wrapper = shallow(
+      <InPanelAlert userName="test" bureau="test" />,
+    );
+    expect(wrapper).toBeDefined();
+  });
+
+  it('matches snapshot', () => {
+    const wrapper = shallow(
+      <InPanelAlert userName="test" bureau="test" />,
```
Comment (burgwyn) [inline src/Components/BidTracker/OverlayAlert/InPanelAlert/InPanelAlert.test.jsx]: "Test"?!  This is your chance to send Lando Calrissian to Cloud City!

File: src/Components/BidTracker/OverlayAlert/OverlayAlert.test.jsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  it('is defined', () => {
+    const wrapper = shallow(
+      <OverlayAlert {...props} />,
+    );
+    expect(wrapper).toBeDefined();
+  });
+
+  // Snapshot tests for each bid status type,
+  // as well as when the type is not handled within the component's switch.
+  types.forEach((type) => {
```
Comment (burgwyn) [inline src/Components/BidTracker/OverlayAlert/OverlayAlert.test.jsx]: 👍

File: src/Components/BidTracker/BidHelpers.js (hunk 2 above)
Comment (mjoyce91) [reply src/Components/BidTracker/BidHelpers.js]: That's what I get for copying and pasting a switch

File: src/Components/BidTracker/BidTrackerCard/BidTrackerCard.jsx (hunk 3 above)
Comment (mjoyce91) [reply src/Components/BidTracker/BidTrackerCard/BidTrackerCard.jsx]: It's either in #1143 or a PR that I haven't opened yet.

File: src/Components/BidTracker/OverlayAlert/InPanelAlert/InPanelAlert.jsx (hunk 6 above)
Comment (mjoyce91) [reply src/Components/BidTracker/OverlayAlert/InPanelAlert/InPanelAlert.jsx]: 4️⃣

File: src/Components/BidListResultsCard/BidStatus/BidStatus.jsx (hunk 1 above)
Comment (mjoyce91) [reply src/Components/BidListResultsCard/BidStatus/BidStatus.jsx]: Added

**Pull request merged**